torch/_inductor/autotune_process.py

```diff
@@ -166,71 +168,179 @@ class BenchmarkRequest:
     can be done inside the same process since they usually don't cause crash.
     """
 
-    module_path: str  # the path of the module defining the triton kernel
-    module_cache_key: str
-    kernel_name: str  # the kernel name defined in the module
-    grid: List[int]
+    # the kernel name defined in the module
+    kernel_name: str
+    input_tensor_meta: List[TensorMeta]
+    output_tensor_meta: TensorMeta
     extra_args: Dict[str, Any]
-    num_stages: int
-    num_warps: int
 
-    input_tensors: Union["TensorMeta", List["TensorMeta"]]
-    output_tensor: Union["TensorMeta", List["TensorMeta"]]
+    def make_run_fn(
```

Comment:
I think this might benefit from a small doc comment telling which kind of run function it makes